tensorflow/python/training/checkpointable/util.py

```diff
@@ -971,9 +971,7 @@ def assert_existing_objects_matched(self):
     for checkpointable_object in list_objects(self._root_checkpointable):
       # Remove data structures that do not contain any variables from
       # restoration checks.
-      if (isinstance(checkpointable_object,
-                     data_structures.CheckpointableDataStructure) and
-              len(checkpointable_object.variables) == 0):
+      if not checkpointable_object._checkpoint_dependencies:
```

Comment:
I'd still limit it to data_structures.CheckpointableDataStructure. Even if something doesn't have dependencies, it's still an "existing object", and could have data associated with itself.